sdk/lib/backup/Backups.ts

```diff
@@ -41,137 +36,141 @@ export type BackupSet<Volumes extends string> = {
+    }>,
+  ) {
+    return this.addSync({
+      dataPath: `/media/startos/volumes/${volume}/` as const,
+      backupPath: `/media/startos/backup/${volume}/` as const,
+      ...options,
+    })
+  }
+  addSync(sync: BackupSync<M["volumes"][0]>) {
+    this.backupSet.push({
+      ...sync,
+      options: { ...this.options, ...sync.options },
+    })
+    return this
+  }
+
+  async createBackup() {
+    for (const item of this.backupSet) {
+      const rsyncResults = await runRsync({
+        srcPath: item.dataPath,
+        dstPath: item.backupPath,
+        options: {
+          ...this.options,
+          ...this.backupOptions,
+          ...item.options,
+          ...item.backupOptions,
+        },
+      })
+      await rsyncResults.wait()
     }
-    const restoreBackup: T.ExpectedExports.restoreBackup = async ({
-      effects,
-    }) => {
-      for (const item of this.backupSet) {
-        const rsyncResults = await runRsync(
-          {
-            dstPath: item.dstPath,
-            dstVolume: item.dstVolume,
-            options: { ...this.options, ...item.options },
-            srcPath: item.srcPath,
-            srcVolume: item.srcVolume,
-          },
-          pathMaker,
-        )
-        await rsyncResults.wait()
-      }
-      return
+    return
+  }
+
+  async restoreBackup() {
+    for (const item of this.backupSet) {
+      const rsyncResults = await runRsync({
+        srcPath: item.backupPath,
+        dstPath: item.dataPath,
+        options: {
+          ...this.options,
+          ...this.backupOptions,
+          ...item.options,
+          ...item.backupOptions,
+        },
+      })
+      await rsyncResults.wait()
     }
-    return { createBackup, restoreBackup }
+    return
   }
 }
-function notEmptyPath(file: string) {
-  return ["", ".", "./"].indexOf(file) === -1
-}
-async function runRsync(
-  rsyncOptions: {
-    srcVolume: string
-    dstVolume: string
-    srcPath: string
-    dstPath: string
-    options: T.BackupOptions
-  },
-  pathMaker: T.PathMaker,
-): Promise<{
+
+async function runRsync(rsyncOptions: {
+  srcPath: string
+  dstPath: string
+  options: T.SyncOptions
+}): Promise<{
   id: () => Promise<string>
   wait: () => Promise<null>
   progress: () => Promise<number>
 }> {
-  const { srcVolume, dstVolume, srcPath, dstPath, options } = rsyncOptions
+  const { srcPath, dstPath, options } = rsyncOptions
 
   const command = "rsync"
   const args: string[] = []
   if (options.delete) {
     args.push("--delete")
   }
-  if (options.force) {
-    args.push("--force")
-  }
-  if (options.ignoreExisting) {
-    args.push("--ignore-existing")
-  }
   for (const exclude of options.exclude) {
     args.push(`--exclude=${exclude}`)
   }
   args.push("-actAXH")
   args.push("--info=progress2")
   args.push("--no-inc-recursive")
-  args.push(pathMaker({ volume: srcVolume, path: srcPath }))
-  args.push(pathMaker({ volume: dstVolume, path: dstPath }))
+  args.push(srcPath)
```

Comment:
Why did you get rid of the path maker, all the paths are going to be wrong are they not. Because they are running outside of the subcontainers and the top level, so all the paths are gonna be wrong. Unless the people are aware they need to have src + dest like /media/containers/main/data

Reply:
the srcPath is generated with the `/media/volumes/...` in the constructor of the BackupSync now